matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/DefaultSession.kt

```diff
@@ -297,4 +307,10 @@ internal class DefaultSession @Inject constructor(
     override fun logDbUsageInfo() {
         RealmDebugTools(realmConfiguration).logInfo("Session")
     }
+
+    override fun launch(block: suspend () -> Unit) {
```

Comment:
Otherwise we could create a sessionScope in the app, and not rely at all on the SDK for this. I'm not sure what is best

Reply:
yes, IDK

Reply:
Seeing as all the uses of `launch` are in the app, I guess it can be avoided?

Reply:
I'm not really confortable with exposing coroutine scope from the sdk. There are 2 problematics we have to tackle here:

1. Makes sure some actions are not cancelled when leaving the screen (so when viewModelScope is closed)
2. Makes sure every suspend methods at the session/room level are cancelled in the SDK when closing session without needing to cancel them in the app 

For the 1 point, I think we can just have some app scope injected (will be a bit like GlobalScope but will be more testable)
For the 2 point, I'm not sure how we should handle this? Any idea @Dominaezzz? using withContext with an inner scope?  Example : 

```
 override suspend fun createRoom(createRoomParams: CreateRoomParams): String {
        return withContext(scopeHolder.scope.coroutineContext){
            createRoomTask.execute(createRoomParams)
        }
    }
```
We have to keep in mind we might have some SDK inner launches too on this scope ^

Reply:
Point 2 is the problematic case. Trying to make the SDK cancel stuff feels a bit artificial, this is something I think apps should take care of.
This is a pretty big design issue though, so it's definitely out of scope of what this PR is trying to fix.
Once everything has been migrated to suspend functions and flows I think it'll be easier to solve point 2. Any solution that is created now would just be something that will have to be fixed later.

Reply:
https://github.com/Kotlin/kotlinx.coroutines/issues/1001 is also worth looking at for the withContext route.